steps/04_supervised_forecasting_api/step.md

```diff
@@ -1,15 +1,165 @@
+`return_pred_int` - whether prediction intervals should be returned  
+`alpha` - prediction interval alpha  
+Returns:  
+`Panel` type object with predictions  
+if `return_pred` is `True`, also returns `Panel` objects that correspond to prediction intervals  
+
+Should we also have  
+`update`  
+`update_predict`  
+?
+
+Tags:
+
+* `requires_fit` - does the model require a call to `fit` before `predict`?
+* `requres_fit_forecast` - does the model require a call to `fit_forecast`?
+* `requires_cutoff` - does the model need a cutoff to specify in `fit`?
+
+## Implementing complex state behaviour
+
+For usability, users can call `predict` directly after `fit`, although it may be more efficient to call `fit_forecast` first.
+
+This can be accomplished by decorating `fit` with a method `fit_forecast_check` that checks whether `fit_forecast` has been called and needs to be called. If has not been called but needs to, runs `fit_forecast` first with arguments passed.    
+
+## Metrics design
+
+Due to the nature of the task, metrics need to compare samples of series to samples of series, i.e., will be of the signature
+
+`mymetric(y_true: Panel, y_pred: Panel, * , sample_weight=None)`
+
+I would suggest to implement metrics not by hand, but by creating one or few decorator factories, that average `sklearn` metrics over time points.
+
+That is, for example, create a function
+
+`make_idx_avg_panel_metric(sklearn_metric : type(mean_squared_error)) -> type(mymetric)`
+
+which creates the panel metric that is the argument metric averaged first over entries in a series, then over the test samples.
+
+To define metrics, define shorthands and create docstrings for the panel metrics created in this way.
+
+## First steps
+
+Suggested first steps:
+
+* implement template class, perhaps without `update`
+* implement decorator `fit_forecast_check` for `fit`
+* write template python file with only gaps to fill in
+* implement "simple example" estimators
+* implement simple metrics - averaged MSE and MAE
+
+
+## Simple examples estimators to implement
+
+* reduction to tabular supervised regression: splits `X` in `fit` into two, by cut-off - past becomes `X`, future becomes `y` for an `sklearn` supervised learner
+* reduction to tabular supervised time series regression: splits `X` in `fit` into two, by cut-off - past becomes `X`, future becomes `y`, or a time series regressor
+* "apply forecaster on the test set by row" - does not require `fit`, and applies copies o the wrapped forecaster to each sample in the prediction set
+* some linear mixed effects model from `statsmodels`
+
+## Notes from 1st meeting
 
 Starting point:
+
 * implement reduction to tabular learning
 
 First models to implement:
+
 * reduction to supervised tabular
 * mixed effects models
 * reduction 2-step: use only constant variables for mean then another estimator for forecast
 
 Other resources
+
 * Master thesis (to be uploaded)
 * previous design notes (will see if I can find any)

```

Comment:
We can remove this, you mention all of these points in more detail above
```suggestion
```

Reply:
sure - just thought to keep it for completeness

Reply:
But they will be in the git history, no?